test/UnitTests/MSTest.Analyzers.UnitTests/TestClassShouldHaveTestMethodAnalyzerTests.cs

```diff
@@ -101,4 +101,44 @@ public static class {|#0:MyTestClass|}
                 .WithLocation(0)
                 .WithArguments("MyTestClass"));
     }
+
+    public async Task WhenTestClassWithoutAssemblyAttributes_InheritsFromClassHasTestMethod_NoDiagnostic()
+    {
+        var code = """
+            using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+            public abstract class BaseClass()
+            {
+                [TestMethod]
+                public void TestMethod1UsingName() { }
+            }
+
+            [TestClass]
+            public class Derived : BaseClass
+            {
+            }
+            """;
+        await VerifyCS.VerifyAnalyzerAsync(code);
+    }
+
+    public async Task WhenTestClassWithoutAssemblyAttributes_InheritsFromClassDoesNotHaveTestMethod_NoDiagnostic()
```

Comment:
Please add another test where the base (and another with base base) contains only assembly init or cleanup to provde we still raise.